File: server/python/app/infrastructure/mappers/payment_details/reattempt_instructions_conditions_mapper.py
from typing import Optional
from app.application.domain.payments.payment_details.reattempt_instructions_conditions import ReattemptInstructionsConditions
from onlinepayments.sdk.domain.reattempt_instructions_conditions import ReattemptInstructionsConditions as ReattemptInstructionsConditionsSdk

class ReattemptInstructionsConditionsMapper:

    @staticmethod
    def map_from_sdk_response(response: Optional[ReattemptInstructionsConditionsSdk]) -> Optional[ReattemptInstructionsConditions]:
        if response is None:
            return None

        dto = ReattemptInstructionsConditions()
        dto.max_attempts = response.max_attempts
        dto.max_delay = response.max_delay

        return dto

File: server/ruby/app/infrastructure/mappers/payment_details/reattempt_instructions_conditions_mapper.rb
module Infrastructure
  module Mappers
    module PaymentDetails
      class ReattemptInstructionsConditionsMapper
        def self.from_sdk_response(response)
          return nil if response.nil?

          Business::Domain::Payments::PaymentDetails::ReattemptInstructionsConditions.new(
            max_attempts: response.max_attempts,
            max_delay: response.max_delay
          )
        end
      end
    end
  end
end